arch/arm/core/cortex_m/tz/CMakeLists.txt

```diff
@@ -12,12 +12,22 @@ if(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
 # when building a Non-Secure image which shall have access to Secure
 # Entry functions.
   zephyr_ld_options(
-    ${LINKERFLAGPREFIX},--out-implib=${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
+    ${LINKERFLAGPREFIX},--out-implib=${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
   )
 
   zephyr_ld_options(
     ${LINKERFLAGPREFIX},--cmse-implib
   )
+
+  set_property(
+    GLOBAL APPEND PROPERTY
+    extra_post_build_byproducts
+    ${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
+    )
 endif()
 
-zephyr_sources(arm_core_tz.c)
+zephyr_link_libraries_ifdef(CONFIG_ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS
```

Comment:
Would you be able to add some inline documentation, here, describing why we need this?

Reply:
Sure. Added some comments now.